p2p/test/basichost/basic_host_test.go

```diff
@@ -199,3 +203,72 @@ func TestAddrFactorCertHashAppend(t *testing.T) {
 		return hasWebRTC && hasWebTransport
 	}, 5*time.Second, 100*time.Millisecond)
 }
+
+func TestWebRTCDirectDialDelay(t *testing.T) {
```

Comment:
I don't understand this test. Is it trying to make sure that if many peers are connected to a server (via QUIC?), that another peer can also connect to that server via WebRTC?

Reply:
There are two tests. 
1. TestOnlyWebRTCDirectDialNoDelay: Tests that dials to a peer with only webrtc address aren't delayed. 
2. TestWebRTCWithManyQUICConnections: Tests that in presence of quic connections webrtc dials are actually delayed. In case there's no happy eyeballs, the 200 initial dials would have taken up all the goroutines available for handshakes in the webrtc listener.

Reply:
>  In case there's no happy eyeballs, the 200 initial dials would have taken up all the goroutines available for handshakes in the webrtc listener.

But you wait for the Connect to finish, so wouldn't the handshake be complete at this point?

Reply:
Yes, the QUIC handshake will complete not the webrtc one. The webrtc handshake will be cancelled. As there's no way to signal this cancellation in webrtc, the server will have to wait for timeout on this handshake. So, at this point if we get a new webrtc handshake attempt, it'll fail because there were no free listener goroutines. This all happens when there's no happy eyeballs. 

With happy eyeballs, the quic handshake should complete without any webrtc dials.